src/main/java/edu/texas/social_computing/hospitals/HRPP.java

```diff
@@ -31,19 +31,28 @@ public static void run(HospitalTable hospitalTable, ResidentTable residentTable)
             // try to match that person
             matching = HRP.run(hospitalTable, residentTable, unmatchedQueue);
             // if matched -> good
-            if(matching.hasAssignment(ndResident)) {
+            if (matching.hasAssignment(ndResident)) {
                 continue;
             }
-        }
-    }
-
-        // Task 2
-        // if not matched
 
+            // Task 2
+            // if partners still not proximally matched
             // unmatch BOTH in the couple and add both back to the queue (now no one is dominant)
+            matching.unassign(ndResident);
+            matching.unassign(partner);
+
+            // ndResident should still be in queue, right???
```

Comment:
I believe that is correct. We'll have to note to make sure that is how @abruck1 does it. This refers to unmatched queue for reference.

Reply:
Yes, I agree. Good call.

Reply:
Done.